projects/libs/core-library-typescript/src/common-library/signatures.ts

```diff
@@ -0,0 +1,65 @@
+import * as crypto from "crypto";
+import * as stringify from "json-stable-stringify";
+import * as fs from "fs";
+import * as _ from "lodash";
+
+export interface SignaturesConfig {
+  publicKeysPath?: string;
+  privateKeyPath?: string;
+}
+
+export class Signatures {
+  config: SignaturesConfig;
+  SIGNATURE_FORMAT: crypto.HexBase64Latin1Encoding = "base64";
+  SIGNATURE_TYPE = "sha256";
+
+  private publicKeys = new Map<string, string>();
+  private privateKey: string;
+
+  public constructor(signaturesConfig: SignaturesConfig) {
+    this.config = signaturesConfig;
+
+    this.readPrivateKey();
+    this.readPublicKeys();
+  }
+
+  private readPrivateKey() {
+    if (this.config.privateKeyPath) {
+      this.privateKey = fs.readFileSync(this.config.privateKeyPath).toString();
+    }
+  }
+
+  private readPublicKeys() {
+    if (this.config.publicKeysPath) {
```

Comment:
What happens if `this.config. publicKeysPath ` is falsy? Shouldn't it error?

Reply:
It's an optional parameter. KeyManager is designed in a way that allows it both sign and validate signatures without forcing it to have keys unnecessary for the task.